src/base/emu-i386/simx86/memory.c

```diff
@@ -356,6 +356,12 @@ int e_handle_pagefault(sigcontext_t *scp)
 			e_printf("DATA node hit at %08x\n",addr);
 		}
 	}
+	/* the page is not unprotected here, the code
+	 * linked by Cpatch will do it */
+	/* ACH: we can set up a data patch for code
+	 * which has not yet been executed! */
+	if (InCompiledCode && Cpatch(scp))
```

Comment:
Is this the right place for it?
I see the block above that, evaluates `p`
variable, and the block below - uses it.
Maybe putting that to line 315 will make
the code more readable?

Reply:
it's the same place as before acee39e. The block above it is for debugging so should be used before Cpatch -- moving it to 315 just makes the debug logs shorter.